src/plugins/README.md

```diff
@@ -0,0 +1,407 @@
+Impress.js Plugins documentation
```

Comment:
This is the current plugins/README.md from my fork. It refers to things not yet possible in this PR, but I thought I'd rather not start editing it into smaller pieces. This also gives you the full picture, even if some plugin types will be introduced in a later commit.

Reply:
OK. We are merging to dev branch anyway, so there is no risk of confusing external people at this point.

It looks like a nice and detailed documentation, thanks.
I don't have time to read it all now, but will try to have a closer look later. (but we don't need to address it in this PR).